# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../bincimap/bincimap-1.3.4-macandmime.patch
# Copyright (C) 2004 - 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Please try this patch and see if it helps.
Jerry Lundström

Index: bincimap-1.3.4/src/mime-printdoc.cc
===================================================================
--- bincimap-1.3.4/src/mime-printdoc.cc	(revision 347)
+++ bincimap-1.3.4/src/mime-printdoc.cc	(working copy)
@@ -58,7 +58,7 @@
   }
 
   mimeSource->reset();
-  mimeSource->seek(headerstartoffsetcrlf);
+  mimeSource->seek(headerstartoffsetcrlf + startoffset);
 
   char c;
   for (unsigned int i = 0; i < length; ++i) {
